aws/acm.go

```diff
@@ -44,25 +67,41 @@ type certificateCache struct {
 	acmClient      acmiface.ACMAPI
 	acmCertSummary []*acm.CertificateSummary
 	acmCertDetail  []*acm.CertificateDetail
+	iamClient      iamiface.IAMAPI
+	iamCertSummary []*iam.ServerCertificateMetadata
+	iamCertDetail  []*iam.ServerCertificate
 }
 
-func newCertCache(acmClient acmiface.ACMAPI) *certificateCache {
+func newCertCache(acmClient acmiface.ACMAPI, iamClient iamiface.IAMAPI) *certificateCache {
 	return &certificateCache{
 		acmClient:      acmClient,
 		acmCertSummary: make([]*acm.CertificateSummary, 0),
 		acmCertDetail:  make([]*acm.CertificateDetail, 0),
+		iamClient:      iamClient,
+		iamCertSummary: make([]*iam.ServerCertificateMetadata, 0),
+		iamCertDetail:  make([]*iam.ServerCertificate, 0),
 	}
 }
 
 func (cc *certificateCache) backgroundCertCacheUpdate(certUpdateInterval time.Duration) {
 	go func() {
 		for {
-			time.Sleep(certUpdateInterval)
 			if err := cc.updateCertCache(); err != nil {
 				log.Printf("Cert cache update failed, caused by: %v", err)
 			} else {
 				log.Println("Successfully updated cert cache")
 			}
+			time.Sleep(certUpdateInterval)
```

Comment:
Yes it was exactly like @lmineiro wrote. We first do a sync to cache and if successful we start the background goroutine to update and enter the main loop.